### PR TITLE
Deprecate existing credit card functions, replaced with new billing functions

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Deprecate existing billing functions (including referral credit card creation)
- Replace with newer billing functions with support for both bank account and credit cards
- Delete old unit tests and cassettes

# Testing

- All unit tests pass as expected, no new recordings needed
- All prior unit tests still work when copy-pasted in, suggesting deprecation was done successfully (no breaking changes)
- Unfortunately, some of the replacement billing functions have to have different function names, since the signature (name + parameter types and order) are the same as the deprecated functions

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by jchen293 — COMMENTED** [review]
Code looks good, was this E2E test with and without a payment method?

File: src/main/java/com/easypost/model/Billing.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+public final class Billing extends EasyPostResource {
+    /**
+     * Delete a payment method.
+     *
+     * @param priority Which type of payment method to delete.
+     * @return true if the payment method was deleted successfully, false otherwise.
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply src/main/java/com/easypost/model/Billing.java:15]: Good catch. Honestly, returning true or throwing an error is not really any different than just returning void or throwing an error. Do we want to return anything from this function?

[the remark this answers is not in the document]
Comment (jchen293) [reply src/main/java/com/easypost/model/Billing.java:15]: Yes, we have determined to return `true` for the endpoint that returns the empty response (we do have a ticket in the backlog)

[the remark this answers is not in the document]
Comment (nwithan8) [reply src/main/java/com/easypost/model/Billing.java]: Fixed

File: src/main/java/com/easypost/model/beta/Referral.java [hunk 2]
```diff
@@ -153,10 +174,62 @@ public static CreditCard addCreditCard(String referralApiKey, String number, int
...
+
+        if (priorityEnum == null) {
+            throw new Exception("Invalid credit card priority.");
+        }
+
+        PaymentMethodObject paymentMethodObject =
+                addCreditCardToUser(referralApiKey, number, expirationMonth, expirationYear, cvc, priorityEnum);
+
+        // Convert the new PaymentMethodObject back into the deprecated CreditCard object
+
```
Comment (Justintime50) [inline src/main/java/com/easypost/model/beta/Referral.java:204]: Nit: no line breaks between a comment and its associated code, leads to confusion.

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Code is good. Per our deprecation documentation though, we need to add a version number in the deprecation comments signifying when this will be removed for ourselves and users to reference. Per our version and docs, we could add "will be removed in v7"

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,11 @@
     - `getSmartrates()` -> `smartrates()`
     - `getLowestSmartRate()` -> `findLowestSmartrate()`
 - Adds OS specific details to the user-agent header
+- Adds `Billing.retrievePaymentMethods()`, `Billing.fundWallet()`, and `Billing.deletePaymentMethod()` functions
+- Deprecates some existing payment method and credit card functions and features:
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply CHANGELOG.md:11]: Resolved

File: src/main/java/com/easypost/model/beta/Referral.java (hunk 2 above)
Comment (nwithan8) [reply src/main/java/com/easypost/model/beta/Referral.java:204]: Resolved

**Comment by nwithan8** [comment]
> Code is good. Per our deprecation documentation though, we need to add a version number in the deprecation comments signifying when this will be removed for ourselves and users to reference. Per our version and docs, we could add "will be removed in v7"

We're on v4 right now. Per our deprecation policy, this should exist for the lifetime of v4 and v5, and be removed in v6.

**Comment by Justintime50** [comment]
> We're on v4 right now. Per our deprecation policy, this should exist for the lifetime of v4 and v5, and be removed in v6.

We are most definitely on v5.5.0 right now. Did you branch off the old v4 tag instead of master?

**Comment by nwithan8** [comment]
> > We're on v4 right now. Per our deprecation policy, this should exist for the lifetime of v4 and v5, and be removed in v6.
> 
> We are most definitely on v5.5.0 right now. Did you branch off the old v4 tag instead of master?

That's my bad. No, I looked at our repo homepage and saw "v4.0.5" as the most recent release, but that's was the backport. Miscommunication, will fix.

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**